### PR TITLE
added new functional

### PR DESCRIPTION
**Pull request by JaysonJuza**
created a Spring Boot project with pom.xml file with all required dependencies
added Book entity
added BookRepository
added BookService
customized database access
customizedcheckstyle
customized application.properties in the test/resources folder
customized CI (Continuous Integration)

### CODE REVIEW

File: mvnw [hunk 1]
```diff
@@ -0,0 +1,308 @@
+#!/bin/sh
```
Comment (MAMentorFX) [inline mvnw]: mvnw and mvnw.cmd should  be in gitignore file

File: src/main/java/onlinebookstoreapp/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity(name = "books")
+@Data
```
Comment (MAMentorFX) [inline src/main/java/onlinebookstoreapp/model/Book.java]: Forget abuut @Data for entities, @Data consists of @EqualsAndHashCode and @ToString, those two may lead to Lazy loading exception. Use @Getter and @Setter instead

File: src/main/java/onlinebookstoreapp/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false)
+    private BigDecimal isbn;
```
Comment (MAMentorFX) [inline src/main/java/onlinebookstoreapp/model/Book.java:24]: shoudl be unique

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

File: mvnw (hunk 1 above)
Comment (MAMentorFX) [reply mvnw]: Not fixed.
Also .mvn/wrapper in gitignore.

File: src/main/java/onlinebookstoreapp/model/Book.java (hunk 3 above)
Comment (MAMentorFX) [reply src/main/java/onlinebookstoreapp/model/Book.java:24]: fixed

**Review by MAMentorFX — CHANGES_REQUESTED** [review]
Put mvn related files to gitignore

**Review by MentorMA — CHANGES_REQUESTED** [review]
resolve problem with git ignore files, clear redundant files from repository

**Review by MAMentorFX — APPROVED** [review]

**Pull request merged**